mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -114,6 +114,10 @@ public void preCreateTable(org.apache.hadoop.hive.metastore.api.Table hmsTable)
       hmsTable.getParameters().put(InputFormatConfig.EXTERNAL_TABLE_PURGE, "TRUE");
     }
 
+    // For a table created by Hive DDL to be readable by Impala, we need the Input and OutputFormat set explicitly
+    hmsTable.getSd().setInputFormat(HiveIcebergInputFormat.class.getCanonicalName());
+    hmsTable.getSd().setOutputFormat(HiveIcebergOutputFormat.class.getCanonicalName());
+
```

Comment:
These are going to be overwritten by `HiveTableOperations` depending on whether Hive is enabled. I think the right fix is to set the property to enable Hive on the table when creating an Iceberg table with the meta hook. @pvary, what do you think?

Reply:
Discussed this with @marton-bod.

After @boroknagyz's PR (#1751) `hive.engine.enabled` drives the value for these properties for HiveCatalog tables.

For Hive tables above Iceberg tables stored in other Iceberg Catalogs these values are still not set. So it might worth to move this change for the non-HiveCatalog codepath here:
https://github.com/apache/iceberg/blob/05752965f5eb453d895db44fb2d072d270646644/mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java#L78-L81

This would allow Impala to read tables created through Hive but stored in non-HiveCatalogs.

For tables created from Hive and stored in HiveCatalog we set `hive.engine.enabled` to `true` here:
https://github.com/apache/iceberg/blob/05752965f5eb453d895db44fb2d072d270646644/mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java#L133-L137

Setting `hive.engine.enabled` for other Iceberg Catalogs does not have any affect, so I think we are good there.

Thanks,
Peter